src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -395,8 +403,8 @@ internal void DisposeWriter()
         /// </summary>
         internal void DisposeAll()
         {
-            writer?.Dispose();
-            dirReader?.Dispose();
+            DisposeWriter();
```

Comment:
It looks like this was causing the regressions, because writer was disposed first so we need to set it to null, otherwise in the `DirReader` getter, we will be trying to access an already disposed object